### PR TITLE
Home responsive: SideDrawer Menu was added

### PR DESCRIPTION
**Pull request by elinaKM**

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -1,33 +1,13 @@
 import React from 'react'
 import styled from 'styled-components'
+import HamMenuControl from './header/HamMenuControl'
 
-const Header = () => (
+const Header = ({ setShowMenu }) => (
```
Comment (okvv) [inline src/components/Header.js]: what's that?
maybe isMenuHidden? toggleMenuVisibility?

File: src/components/sideMenu/SideDrawerMenu.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import styled from 'styled-components'
+import SideMenuCloseControl from './SideMenuCloseControl'
+import { css } from 'styled-components'
+
+const SideDrawerMenu = ({open, onCloseMenu}) => (
```
Comment (okvv) [inline src/components/sideMenu/SideDrawerMenu.js]: isOpen

File: src/pages/Home.js [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
+import Backdrop from './../components/common/Backdrop'
 
-const Home = () => (
-    <Wrapper>
-        <Wellcome/>
-        <Footer/>
-    </Wrapper>
-)
+const Home = () => {
+    const [showMenu, setShowMenu] = useState(false);
```
Comment (okvv) [inline src/pages/Home.js]: isMenuVisible?

Comment (okvv) [reply src/pages/Home.js]: toggleMenuVisiblity? / toggleMenu?

File: src/components/sideMenu/SideDrawerMenu.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    <Wrapper open = {open}>
+        <SideMenuCloseControl onClick = {onCloseMenu}/>
```
Comment (okvv) [inline src/components/sideMenu/SideDrawerMenu.js]: what happens onClick? closeMenu? or toggleMenu?
try to read the code like a conversation

File: src/components/header/HamMenuControl.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import styled from 'styled-components'
+import menuIcon from '../../images/icons/ham-menu.svg'
+
+const HamMenuControl = ({ onClick }) => (
+  <Wrapper onClick={onClick}>
```
Comment (okvv) [inline src/components/header/HamMenuControl.js]: onClick={clickHandler } to prevent confusion

File: src/components/common/Backdrop.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import styled from 'styled-components'
+
+const Backdrop = ({onClick}) => <Wrapper onClick={onClick}/>
```
Comment (okvv) [inline src/components/common/Backdrop.js]: onClick={clickHandler } to prevent confusion

File: src/components/Wellcome.js [hunk 7]
```diff
@@ -4,23 +4,23 @@ import dogBackgroundImg from '../images/dog-background.jpg'
 import Header from './Header'
 import Button from './common/Button'
 
-const Wellcome = () => (
+const Wellcome = ({ setShowMenu }) => (
```
Comment (okvv) [inline src/components/Wellcome.js]: toggleMenuVisibility

**Review by okvv — CHANGES_REQUESTED** [review]

File: src/pages/Home.js (hunk 3 above)
Comment (elinaKM) [reply src/pages/Home.js]: done

File: src/components/sideMenu/SideDrawerMenu.js (hunk 2 above)
Comment (elinaKM) [reply src/components/sideMenu/SideDrawerMenu.js]: done

File: src/components/common/Backdrop.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import styled from 'styled-components'
+
+const Backdrop = ({clickHandler }) => <Wrapper onClick={clickHandler }/>
```
Comment (okvv) [inline src/components/common/Backdrop.js:4]: click handler is too general, you'll probably use it like this: onClick=clickHandler which tells me nothing about what clicking does

File: src/components/Wellcome.js [hunk 9]
```diff
@@ -4,23 +4,23 @@ import dogBackgroundImg from '../images/dog-background.jpg'
...
                 <Title>
                     Find your Best Friend
                 </Title>
                 <Text>
-                    This site is a perfect place to find a perfect pet.<br/>
-                    Animals bring a lot of light and joy in our life.<br/>
-                    Pick one.<br/>
-                    Save one live.<br/>
-                    Find a new love.<br/>
+                    This site is a perfect place to find a perfect pet.<br />
```
Comment (okvv) [inline src/components/Wellcome.js:19]: FYI: you can make a new component and pass text as a reference
anyway I don't think BR scales well on multiple screen and with any length of text

leave it be for the time being

File: src/pages/Home.js [hunk 10]
```diff
@@ -1,23 +1,29 @@
...
-        <Wellcome/>
-        <Footer/>
-    </Wrapper>
-)
+const Home = () => {
+    const [isMenuVisible, setMenuVisible] = useState(false);
+
+    return (
+        <Wrapper>
+            <SideDrawerMenu isOpen={isMenuVisible} onCloseMenu={() => setMenuVisible(false)}/>
```
Comment (okvv) [inline src/pages/Home.js]: as long as you're aware of the rendering costs of passing functions like this it's ok

**Review by okvv — APPROVED** [review]
please read comments

File: src/pages/Home.js [hunk 11]
```diff
@@ -1,23 +1,32 @@
...
+
+const Home = () => {
+    const [isMenuVisible, setMenuVisible] = useState(false);
+    
+    const sideDrawerCloseHandler = () => setMenuVisible(false);
+
+    return (
+        <Wrapper>
+            <SideDrawerMenu isOpen={isMenuVisible} onCloseMenu={sideDrawerCloseHandler}/>
+            { isMenuVisible && <Backdrop clickHandler ={sideDrawerCloseHandler}/>}
```
Comment (okvv) [inline src/pages/Home.js]: space

**Review by okvv — APPROVED** [review]

**Comment by okvv** [comment]
review comment and merge

**Comment by elinaKM** [comment]
merge

**Pull request merged**